code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -631,11 +631,11 @@
 		// some upper limits
 		if(shards > max_ex_shards)
 			shards = max_ex_shards
-		if(istype(shard_type, /datum/ammo/bullet/shrapnel/incendiary) && shards > max_ex_shards / INCENDIARY_SHARDS_MAX_REDUCTION) // less max incendiary shards
+		if(ispath(shard_type, /datum/ammo/bullet/shrapnel/incendiary) && shards > max_ex_shards / INCENDIARY_SHARDS_MAX_REDUCTION) // less max incendiary shards
```

Comment:
i think you can remove all of this and instead just add 
max_ex_shards = max_exshards / INCENDIARY_SHARDS_MAX_REDUCTION 

you can repeat that process in line 624 , 626 and 628 . so there is no need to check twice and its easier to read